docs: add v18 migration guide (#4378)
ngrx/platform: projects/ngrx.io/content/guide/migration/v18.md

# V18 Update Guide

## Angular CLI update

> NgRx supports using the Angular CLI `ng update` command to update your dependencies. Migration schematics are run to make the upgrade smoother. These schematics will fix some of the breaking changes.

Reviewer: ```suggestion
NgRx supports using the Angular CLI `ng update` command to update your dependencies. Migration schematics are run to make the upgrade smoother. These schematics fix some of the breaking changes.
```

To update your packages to the latest released version, run the command below.

```sh
ng update @ngrx/store@18
```

## Dependencies

Version 18 has the minimum version requirements:

- Angular version 18.x
- Angular CLI version 18.x

> - TypeScript version 5.x

Reviewer: ```suggestion
- TypeScript version 5.4.x
```

- RxJS version ^6.5.x || ^7.5.0

## Breaking changes

### @ngrx/store

#### Merge `Action` and `TypedAction` interfaces

The Action and TypedAction interfaces are merged into one interface.

> This change has a ng-update schematic that will automatically update your code.

Reviewer: ```suggestion
This change has an ng-update schematic that automatically updates your code.
```

BEFORE:

There was a separation between the Action and TypedAction interfaces.

AFTER:

The Action interface accepts a generic type parameter that represents the payload type (defaults to string).
The TypedAction interface is removed.

### @ngrx/effects

#### Remove `concatLatestFrom` operator

The `concatLatestFrom` operator has been removed from `@ngrx/effects` in favor of the `@ngrx/operators` package.

> This change has a ng-update schematic that will automatically update your code.

Reviewer: ```suggestion
This change has an ng-update schematic that automatically updates your code.
```

BEFORE:

```ts
import { concatLatestFrom } from '@ngrx/effects';
```

AFTER:

```ts
import { concatLatestFrom } from '@ngrx/operators';
```

### @ngrx/component-store

#### Remove `tapResponse` operator

The `tapResponse` operator has been removed from `@ngrx/component-store` in favor of the `@ngrx/operators` package.

> This change has a ng-update schematic that will automatically update your code.

Reviewer: ```suggestion
This change has an ng-update schematic that automatically updates your code.
```
